fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Channel.java

```diff
@@ -0,0 +1,132 @@
+package com.github.dedis.popstellar.model.objects;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class represent a channel used to subscribe to a certain part of the messages. More detailed
+ * information are given in the protocol specification of the application.
+ *
+ * <p>It is implemented in a functional manner to avoid any unintended modification
+ */
+public class Channel {
+
+  private static final String ROOT_CHANNEL = "/root";
+  private static final String DELIMITER = "/";
+  // Predicate matching a String with the protocol regex of a channel
+  private static final Predicate<String> CHANNEL_VALIDATOR =
+      Pattern.compile("^/root(/[^/]+)*$").asPredicate();
+
+  /** Root channel, every client is subscribed to it. */
+  public static final Channel ROOT = new Channel();
+
+  public static Channel newChannel(String value) {
+    // Avoid creating duplicate instances
+    if (ROOT_CHANNEL.equals(value)) return ROOT;
```

Comment:
this looks like an ad-hoc version of the flyweight pattern 😅 
Should we go full flyweight ? I'm not sure if it's needed.

Reply:
I might have overdone things a little.

I did it only for the Root channel because I already had a defined constant, and it is easy to isolate.
But for clarity, it is probably better to remove that part.